homeassistant/components/starline/account.py

```diff
@@ -173,6 +173,9 @@ def engine_attrs(device: StarlineDevice) -> dict[str, Any]:
         return {
             "autostart": device.car_state.get("r_start"),
             "ignition": device.car_state.get("run"),
+            "handsfree": device.car_state.get("hfree"),
+            "neutral": device.car_state.get("neutral"),
+            "moving_ban": device.car_state.get("arm_moving_pb"),
```

Comment:
What do these values represent? If they are relevant on their own they should be separate sensor entities.

Reply:
handsfree - open the car when you are nearby and close it if you move away;
neutral - program neutral status for autostart car with manual transmission;
moving_ban - status of moving ban (you can't use car)

These are not sensors, because these values do not change on their own. This is the status of the starline alarm system settings. Needed to display states in the starline-lovelace-card

Reply:
What's the value type for each of them? A boolean?

If they can never change they should not be in the state machine at all. If they can change, it sounds like they should be binary sensors if they are booleans, otherwise separate sensors.

Reply:
It doesn't matter if you only want these states for a custom GUI. We require them to be separate entities if they fulfill the criteria of meaning something on their own, which in this case they do. Please move them to separate binary sensors entities.

Reply:
What about another attributes: autostart, ignition? Should i move them to separate binary sensor entities?

Reply:
Yes, but it's not a requirement right now as it will be a breaking change to remove them from the state attributes. I'd recommend doing it at some point now or later.

Reply:
Do you even understand how a car security system works? How does starline work?
These values are not independent; they are attributes of the security system. They cannot be changed; these values have no meaning on their own.
Logically, these are exactly attributes.
Please consider the context when reviewing.

Reply:
As I understand it, the `handsfree` value represents the system setting on/off for the feature that opens the car when you get close and closes the car if you move away. This setting does have a meaning on its own. Have I misunderstood something in your explanation?

Reply:
There is difference between starline data values. For example, door state is binary sensor, because you can open door and it's state of car. Battery percent, engine temperature are sensors.
But security system have a lot of settings that you can change by mobile app, but usually these settings are enabled or disabled once. For example, if you are using handsfree - enable it once. If your car needs programmable neitral for autostart - enable it once. These values are not "sensors" in my opinion (and not only my, we discudssed about integration improvements with our community), because history of these values are not important. These values are not triggers for automations, just attributes. But it's important to have ability to check these values.

Look at the picture. There are autostart and handsfree icons there.
![9HsXlo2ztdg](https://github.com/home-assistant/core/assets/1973755/dfdb9e79-9ba6-4a87-aaba-b2dd1e3eca0f)

Starline integration makes for about 20 entities for each car. I think, it's not needed to create more, especially for immutable values.

Reply:
Home Assistant doesn't make the distinction that you describe between the sensors on the car and the configuration settings of the system. Both can and should be sensors if they are to be exposed. They way we can make a distinction is to set the entity category attribute, if it's appropriate.

Reply:
Then what are attributes? When can they be used?

Reply:
They add extra information to the main state of the entity. Eg a light entity has the main state on or off. It can have an additional attribute for the brightness. The brightness information isn't useful without knowing if the light is on or off.